src/System.Text.Json/src/System/Text/Json/Node/JsonArray.cs

```diff
@@ -0,0 +1,675 @@
+        /// <param name="item">The item to add.</param>
+        /// <exception cref="ArgumentException">
+        ///   Provided value is not in a legal JSON number format.
+        /// </exception>
+        public void Insert(int index, double item) => Insert(index, new JsonNumber(item));
+
+        /// <summary>
+        ///   Inserts the specified item as a <see cref="JsonNumber"/> at the specified index of the JSON array.
+        /// </summary>
+        /// <param name="index">The zero-based index at which item should be inserted.</param>
+        /// <param name="item">The item to add.</param>
+        [CLSCompliant(false)]
+        public void Insert(int index, sbyte item) => Insert(index, new JsonNumber(item));
+
+        /// <summary>
+        ///   Inserts the specified item as a <see cref="JsonNumber"/> at the specified index of the JSON array.
+        /// </summary>
+        /// <param name="index">The zero-based index at which item should be inserted.</param>
+        /// <param name="item">The item to add.</param>
+        [CLSCompliant(false)]
+        public void Insert(int index, ushort item) => Insert(index, new JsonNumber(item));
+
+        /// <summary>
+        ///   Inserts the specified item as a <see cref="JsonNumber"/> at the specified index of the JSON array.
+        /// </summary>
+        /// <param name="index">The zero-based index at which item should be inserted.</param>
+        /// <param name="item">The item to add.</param>
+        [CLSCompliant(false)]
+        public void Insert(int index, uint item) => Insert(index, new JsonNumber(item));
+
+        /// <summary>
+        ///   Inserts the specified item as a <see cref="JsonNumber"/> at the specified index of the JSON array.
+        /// </summary>
+        /// <param name="index">The zero-based index at which item should be inserted.</param>
+        /// <param name="item">The item to add.</param>
+        [CLSCompliant(false)]
+        public void Insert(int index, ulong item) => Insert(index, new JsonNumber(item));
+
+        /// <summary>
+        ///   Inserts the specified item as a <see cref="JsonNumber"/> at the specified index of the JSON array.
+        /// </summary>
+        /// <param name="index">The zero-based index at which item should be inserted.</param>
+        /// <param name="item">The item to add.</param>
+        public void Insert(int index, decimal item) => Insert(index, new JsonNumber(item));
+
+        /// <summary>
+        ///   Determines whether a specified <see cref="JsonNode"/> element is in a JSON array.
+        /// </summary>
+        /// <param name="value">Value to check.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the value is successfully found in a JSON array,
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        public bool Contains(JsonNode value) => _list.Contains(value);
```

Comment:
All of these Contains methods are going to be hard to describe.  I recommend removing them.

(Is it reference equality, or value equality? What does it mean to Contains(5.0f)? `5`? `5.0`? `5.0000000000000000000000000000000000000000000000000000000001`? `51e-1`?

Reply:
:disappointed_relieved:

Reply:
Given that this method (and IndexOf) are doing reference equality, we should delete Contains. IndexOf, and LastIndexOf until we can come up with a stronger definition.

This is made worse by the implicit conversions, because  `array.Add(true)`, `array.Contains(true) => false`.